src/Data/Nat/Primality.agda

```diff
@@ -106,7 +317,8 @@ euclidsLemma m n {p@(suc (suc _))} p-prime p∣m*n = result
   result with Bézout.lemma m p
   -- if the GCD of m and p is zero then p must be zero, which is
   -- impossible as p is a prime
-  ... | Bézout.result 0 g _ = contradiction (0∣⇒≡0 (GCD.gcd∣n g)) λ()
+  ... | Bézout.result 0 g _ = contradiction (0∣⇒≡0 (GCD.gcd∣n g)) (≢-nonZero⁻¹ _)
+  -- this should be a typechecker-rejectable case!?
```

Comment:
Two observations:
* the `contradiction` argument basically reprises that on L285 above, so can we isolate a common abstraction?
* I couldn't (easily) find a way for the in stance on L307 above to rule out this case from even occurring, by unification failure in the typechecker; but I think it *ought* to be possible?

Reply:
Closing this for now. I think it needs some refactoring of the `NonZero` instance reasoning in `GCD`...